packages/react-router/src/link.tsx

```diff
@@ -497,34 +505,34 @@ export function useLinkProps<
 
   const handleEnter = (e: MouseEvent) => {
     if (disabled) return
-    const target = (e.target || {}) as LinkCurrentTargetElement
+    const eventTarget = (e.target || {}) as LinkCurrentTargetElement
 
     if (preload) {
-      if (target.preloadTimeout) {
+      if (eventTarget.preloadTimeout) {
         return
       }
 
-      target.preloadTimeout = setTimeout(() => {
-        target.preloadTimeout = null
+      eventTarget.preloadTimeout = setTimeout(() => {
+        eventTarget.preloadTimeout = null
         doPreload()
       }, preloadDelay)
     }
   }
 
   const handleLeave = (e: MouseEvent) => {
     if (disabled) return
-    const target = (e.target || {}) as LinkCurrentTargetElement
+    const eventTarget = (e.target || {}) as LinkCurrentTargetElement
 
-    if (target.preloadTimeout) {
-      clearTimeout(target.preloadTimeout)
-      target.preloadTimeout = null
+    if (eventTarget.preloadTimeout) {
+      clearTimeout(eventTarget.preloadTimeout)
+      eventTarget.preloadTimeout = null
     }
   }
 
   const composeHandlers =
-    (handlers: (undefined | ((e: any) => void))[]) =>
+    (handlers: Array<undefined | ((e: any) => void)>) =>
     (e: React.SyntheticEvent) => {
-      if (e.persist) e.persist()
+      e.persist()
```

Comment:
On typing level, the event is a `React.SyntheticEvent`, so linting now says that we shouldn't need to check for the persist function before invoking it, because it should be always there.

why do react aria events not use SyntheticEvents, and how does that work please? What happens if we call other function, like `event.preventDefault()`? I don't really think this is an issue on our end, but more on react-aria's side?

Reply:
I do understand why it got removed but I just wanted to point out that there are libraries which might not even emit a `React.SyntheticEvent` so maybe the typing could also be loosened here.

> why do react aria events not use SyntheticEvents

That's a good question I cannot answer as I'm just a user. I was also surprised to find out that React Aria's custom events are plain objects and not real events. There's also no `preventDefault()` on these which has also bitten me. I'll also check on their side but my current situation simply is that preloading is broken and the console is spammed, even if preloading is disabled. Let me apply a patch through `pnpm` to bring the check back for now in my project until this is resolved on either side.

Reply:
see:
- https://github.com/TanStack/router/pull/1443